sql/analyzer/apply_join.go

```diff
@@ -241,3 +241,53 @@ func normalizeSelectSingleRel(ctx *sql.Context, a *Analyzer, n sql.Node, scope *
 		return c.Node, transform.SameTree, nil
 	})
 }
+
+func subqeCanCommute(n sql.Node, scope *Scope, lowestAllowedIdx int) bool {
```

Comment:
(minor) some function docs would help future code readers here. The implementation feels like it's about caching, and I'm guessing there's a little extra context about commuting that would be helpful for future readers. For example, it would be nice to explain why/how this is different than `IsNodeCacheable` so readers don't have to reverse engineer that by comparing the code.